dts/arm/silabs/xg23/efm32pg23b310f512im48.dtsi

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2025 Christoph Jans
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <silabs/xg23/xg23.dtsi>
+
+/ {
+	sram0: memory@20000000 {
```

Comment:
Should be moved to the bottom of the file and updated using a phandle reference for consistency with https://github.com/zephyrproject-rtos/zephyr/pull/93810

i.e.
```
&sram0 {
	reg = <0x20000000 DT_SIZE_K(64)>;
};
```